src/samplers/aliastable.jl

```diff
@@ -1,21 +1,21 @@
-struct AliasTable{S} <: Sampleable{Univariate,Discrete}
-    accept::Vector{Float64}
-    alias::Vector{Int}
-    isampler::S
+struct AliasTable{N} <: Sampleable{Univariate,Discrete}
+    accept::NTuple{N,Float64}
+    alias::NTuple{N,Int}
+    sample_space::UnitRange{Int64}
 end
-ncategories(s::AliasTable) = length(s.accept)
+ncategories(s::AliasTable{N}) where N = N
```

Comment:
Since we'd have to change the representation anyway, I think it's fine to just drop the third field since there is little benefit from storing a `UnitRange` over just constructing it when it's needed.

I'm also wondering you are sure that the change to `NTuple` is right. I'm sympathetic but are we sure that people are using this function with long vectors?

Reply:
> I'm also wondering you are sure that the change to NTuple is right. I'm sympathetic but are we sure that people are using this function with long vectors?

I don't think AliasTable is part of the public API, so this is just an implementation detail, no? I have not changed the type of `probs` - it is still a `Vector`.

Reply:
The concern is that if people are using this with many categories then these tuples might get very large. It's probably better to leave these as vectors for now and have the discussion of the representation in a separate PR.

Reply:
Right. Will fix this.

Reply:
Done

Reply:
Also dropped last field of AliasTable as suggested.